frontend/lib/src/components/widgets/DataFrame/hooks/useColumnLoader.test.ts

```diff
@@ -183,6 +183,142 @@ describe("applyColumnConfig", () => {
+        },
+      ],
+      // 2. Position-based config
+      [
+        `${COLUMN_POSITION_PREFIX}0`,
+        {
+          width: "medium",
+          label: "Position Label",
+          alignment: "center",
+        },
+      ],
+      // 3. Name-based config
+      [
+        "",
+        {
+          width: "large",
+          label: "Name Label",
+          alignment: "right",
+        },
+      ],
+      // 4. ID-based config
+      [
+        "index_col",
+        {
+          width: 100,
+          label: "ID Label",
+          alignment: "left",
+        },
+      ],
+    ])
+
+    // Test with the index column from MOCK_COLUMNS
+    const result = applyColumnConfig(MOCK_COLUMNS[0], columnConfig)
+
+    // Config should be merged in order, with later configs overwriting earlier ones
+    expect(result).toEqual({
+      ...MOCK_COLUMNS[0],
+      // Should have the width from ID config (last)
+      width: 100,
+      // Should have the label from ID config (last)
+      title: "ID Label",
+      // Should have the alignment from ID config (last)
+      contentAlignment: "left",
+    })
+  })
+
+  it("allows partial config overrides in priority order", () => {
+    const columnConfig: Map<string | number, ColumnConfigProps> = new Map([
+      [
+        INDEX_IDENTIFIER,
+        {
+          width: "small",
+          label: "Index Label",
+        },
+      ],
+      [
+        `${COLUMN_POSITION_PREFIX}0`,
+        {
+          // Only override the label
+          label: "Position Label",
+        },
+      ],
+      [
+        "index_col",
+        {
+          // Only override the width
+          width: 100,
+        },
+      ],
+    ])
+
+    const result = applyColumnConfig(MOCK_COLUMNS[0], columnConfig)
+
+    expect(result).toEqual({
+      ...MOCK_COLUMNS[0],
+      // Width should come from ID config
+      width: 100,
+      // Label should come from position config
+      title: "Position Label",
+    })
+  })
+
+  it("correctly merges nested type_config options", () => {
+    const columnConfig: Map<string | number, ColumnConfigProps> = new Map([
+      // 1. Index config
+      [
+        INDEX_IDENTIFIER,
+        {
+          type_config: {
+            options: ["a", "b"],
+            min_value: 0,
+          },
+        },
+      ],
+      // 2. Position-based config
+      [
+        `${COLUMN_POSITION_PREFIX}0`,
+        {
+          type_config: {
+            options: ["c", "d"],
+            max_value: 100,
+          },
+        },
+      ],
+      // 3. ID-based config
+      [
+        "index_col",
+        {
+          type_config: {
+            options: ["e", "f"],
+            step: 1,
+          },
+        },
+      ],
+    ])
+
+    const result = applyColumnConfig(MOCK_COLUMNS[0], columnConfig)
+
+    // Should merge all type_config options from different config sources
+    expect(result.columnTypeOptions).toEqual({
+      options: ["e", "f"], // From ID config (last)
```

Comment:
For my own learning, why do `"e"` and `"f"` show up here but not `"a", "b", "c", "d"`

Reply:
The overall goal is that only object keys are merged by overwritten values. This also includes lists. However, the test was a bit misleading since lodash merge actually tries to merge arrays on a index-by-index base. I replaced it with a customized merge that replaces array values instead of merging them.